include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,505 @@
+                return std::weak_ordering::equivalent;
+            return std::weak_ordering::greater; //x.current > y.current_
+        }
+        
+        friend constexpr auto
+        operator-(const iterator& x, const iterator& y) requires 
+        (std::sized_sentinel_for<std::ranges::iterator_t<__maybe_const<Const, Views>>, std::ranges::iterator_t<__maybe_const<Const, Views>>> && ...)
+        {
+            return y.distance_to_it(x, std::make_index_sequence<sizeof...(Views)>());
+        }
+
+        friend constexpr iterator
+        operator+(iterator it, difference_type n) requires all_random_access<Const, Views...>
+        {
+            return it += n;
+        }
+
+        friend constexpr iterator
+        operator+(difference_type n, iterator it) requires all_random_access<Const, Views...>
+        {
+            return it += n;
+        }
+
+        friend constexpr iterator
+        operator-(iterator it, difference_type n) requires all_random_access<Const, Views...>
+        {
+            return it -= n;
+        }
+
+        friend constexpr decltype(auto) iter_move(const iterator& x) noexcept(
+            (noexcept(std::ranges::iter_move(std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>())) && ...) &&
+            (std::is_nothrow_move_constructible_v<std::ranges::range_rvalue_reference_t<__maybe_const<Const, Views>>> && ...))
+        {
+            auto __tr = [](auto&&... __args) -> decltype(auto) {
+                return rvalue_reference_type(std::forward<decltype(__args)>(__args)...);
+            };
+            return apply_to_tuple(__tr, std::ranges::iter_move, x.current_);
+
+        }
+
+        friend constexpr void iter_swap(const iterator& x, const iterator& y) noexcept(
+            (noexcept(std::ranges::iter_swap(std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>(),
+                                        std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>())) && ...))
+          requires(std::indirectly_swappable<std::ranges::iterator_t<__maybe_const<Const, Views>>> && ...)
+        {
+            bi_apply_to_tuple(std::ranges::iter_swap, x.current_, y.current_);
+        }
+
+      private:
+        template <std::size_t... In>
+        constexpr bool
+        compare_equal(iterator y, std::index_sequence<In...>) const
+        {
+            return ((std::get<In>(current_) == std::get<In>(y.current_)) || ...);
+        }
+
+        template <typename SentinelsTuple, std::size_t... In>
+        constexpr bool
+        compare_with_sentinels(const SentinelsTuple& sentinels, std::index_sequence<In...>) const
+        {
+            return ((std::get<In>(current_) == std::get<In>(sentinels)) || ...);
+        }
+        
+        template <typename SentinelsTuple, std::size_t... In>
+        constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>        
```

Comment:
It seems like we can use difference_type alias here and in other distance functions